### PR TITLE
Added download csv options for enrollments and attendance

### PR DESCRIPTION
**Pull request by aoby**
## Enrollments:

Added 
image download button:

image
## Attendance:

Added `Download CSV` button, changed `Cancel` to `Back`, and stopped `Save` from navigating away when it's done. `Save` is disabled when no changes have been made, and `Download CSV` is disabled when changes are unsaved.
image

After modifying:
image
### Note

I also refactored some code in the API that was used for serializing reports to csv into a more general lib module that can be included in controllers for this functionality.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/attendance/workshop_attendance.jsx [hunk 1]
```diff
@@ -204,9 +221,18 @@ const WorkshopAttendance = React.createClass({
...
             <ButtonToolbar>
-              <Button bsStyle="primary" onClick={this.handleSaveClick}>Save</Button>
+              <SplitButton
+                bsStyle="primary"
+                title="Save"
+                id="save-button"
+                onClick={this.handleSaveClick}
+              >
+                <MenuItem eventKey="1" onSelect={this.handleSaveAndDownloadCsvClick}>
+                  Save and download Csv
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/attendance/workshop_attendance.jsx]: csv or CSV

File: dashboard/test/factories.rb [hunk 2]
```diff
@@ -502,6 +507,15 @@
     workshop_type Pd::Workshop::TYPES.first
     course Pd::Workshop::COURSES.first
     capacity 10
+    transient do
+      num_sessions 0
+    end
+    after(:create) do |workshop, evaluator|
+      # Sessions, one per day starting today
```
Comment (mehalshah) [inline dashboard/test/factories.rb:514]: Every day, every weekday, or something else?

**Comment by mehalshah** [comment]
LGTM!

File: dashboard/test/factories.rb (hunk 2 above)
Comment (aoby) [inline dashboard/test/factories.rb:514]: It doesn't matter in this context. I just wanted a factory to create `n` sessions, and they must be on different days based on the model validations. None of the tests care which specific day they are on, and if we ever need that we can use the existing `pd_session` factory to create them directly with a set date.

**Pull request merged**